File: services/nuldoc/nuldoc-src/generators/tag_list.ts
import TagListPage from "../pages/TagListPage.ts";
import { Config } from "../config.ts";
import { Page } from "../page.ts";
import { TagPage } from "./tag.ts";

export type TagListPage = Page;

export async function generateTagListPage(
  tags: TagPage[],
  site: "blog" | "slides",
  config: Config,
): Promise<TagListPage> {
  const html = await TagListPage(tags, site, config);

  return {
    root: html,
    renderer: "html",
    site,
    destFilePath: "/tags/index.html",
    href: "/tags/",
  };
}

File: services/nuldoc/lib/nuldoc/generators/tag_list.rb
module Nuldoc
  module Generators
    class TagList
      def initialize(tags, site, config)
        @tags = tags
        @site = site
        @config = config
      end

      def generate
        html = Pages::TagListPage.new(tags: @tags, site: @site, config: @config).render

        Page.new(
          root: html,
          renderer: :html,
          site: @site,
          dest_file_path: '/tags/index.html',
          href: '/tags/'
        )
      end
    end
  end
end
